pyspecProcScripts/integrate_limits.py

```diff
@@ -1,43 +1,148 @@
+    #{{{ make sure to calculate Lorentzian filter for L-to-G transform
+    if Lorentz_to_Gauss:
+        assert Lorentzian_Conv == True and Gaussian_Conv == False, "Must set Lorentzian convolve to True for Lorentzian to Gaussian transformation"
+    #}}}
+    # Calculating matched filter
+    sigma = nddata(np.linspace(1e-10,1e-1,1000),'sigma').set_units('sigma','s')
+    if Gaussian_Conv:
+        gaussians = np.exp(-temp.C.fromaxis(axis)**2/2/sigma**2)
+        signal_E = (abs(temp * gaussians)**2).sum(axis)
+    if Lorentzian_Conv:
+        lorentzians = np.exp(-abs(temp.C.fromaxis(axis))/sigma)
+        signal_E = (abs(temp * lorentzians)**2).sum(axis)
+    signal_E /= signal_E.data.max()
+    if Gaussian_Conv:
+        filter_width = abs(signal_E-1/sqrt(2)).argmin('sigma').item()
+    if Lorentzian_Conv:
+        filter_width = abs(signal_E-signal_E.max()/2).argmin('sigma').item()
+    if fl is not None:
+        fl.next('integration diagnostic -- signal Energy')
         fl.plot(signal_E, human_units=False)
         fl.plot(signal_E['sigma':(filter_width,filter_width+1e-6)],'o', human_units=False)
+    print("*** *** ***")
+    print("FILTER WIDTH IS",filter_width)
+    print("*** *** ***")
+    # Generate the appropriate convolution functions
+    Gaussian_func = lambda x,y: exp(-(x**2)/(2.0*(y**2)))
+    Lorentzian_func = lambda x,y: exp(-(abs(x))/y)
+    if Gaussian_Conv:
+        convfunc = Gaussian_func
+    if Lorentzian_Conv:
+        convfunc = Lorentzian_func
+    if Lorentz_to_Gauss:
+        # not immediately sure why, but I need to do this in order to get freq limits around the peak (otherwise they are around 0)
+        rough_center = abs(temp).C.mean_all_but('t2').argmax('t2').item()
+        temp.setaxis('t2', lambda t: t- rough_center).register_axis({'t2':0})
+    if Lorentz_to_Gauss:
+        temp = temp['t2':(7e-3,None)]
+    manual_convolve = False
+    if manual_convolve:
+        #{{{ for manual convolution - arguably can be deleted
+        # assumed temp starts in time domain
+        for_manual = temp.C
+        x = for_manual.C.fromaxis('t2')
+        myfilter = convfunc(x,filter_width)
+        if Gaussian_Conv:
+            fl.next('Gaussian filter')
+        if Lorentzian_Conv:
+            fl.next('Gaussian filter')
+        fl.plot(myfilter)
+        fl.show();quit()
+        if Gaussian_Conv:
+            fl.next('Compare Gaussian Filter and Abs Data Normalized')
+        if Lorentzian_Conv:
+            fl.next('Compare Lorentzian Filter and Abs Data Normalized')
+        fl.plot(abs(for_manual)/abs(for_manual).max(), alpha=0.5, label='abs data')
+        fl.plot(myfilter, alpha=0.5, label='filter')
+        newdata = for_manual*myfilter
+        fl.next('Overlay abs data')
+        fl.plot(abs(for_manual), alpha=0.5, label='before applying filter')
+        fl.plot(abs(newdata), alpha=0.5, label='after applying filter')
+        temp = newdata.C
+        temp.ft('t2')
+    #}}}
+    if not manual_convolve:
```

Comment:
I think just delete the manual convolution option

Reply:
I will also just note that this example only works correctly if fix_convolve.py is checked out on pyspecdata, the last that I checked, unless those changes are now on master (and master is up to date on your computer)

Reply:
so, I actually went to merge that one in, and noticed that there are problems with it.

I think you want to go and make the fixes there that I'm suggesting -- then to maintain the behavior of this code, just invert the width in any call to `convolve` that you make here.

Reply:
Yes I think that is good now, and here I will invert the line widths so that it is compatible. I did not merge the convolve branch with master yet, which is what I had done with this one before you reviewed it -- if you want me to do that, just let me know